tests/operator_test.go

```diff
@@ -402,6 +510,173 @@ var _ = Describe("Operator", func() {
+				Expect(err).ToNot(HaveOccurred())
+
+				// Use Current virtctl to start VM
+				// NOTE: we are using virtctl explicitly here because we want to start the VM
+				// using the subresource endpoint in the same way virtctl performs this.
+				By("Starting VM with virtctl")
+				startFn := tests.NewRepeatableVirtctlCommand("start", "--namespace", tests.NamespaceTestDefault, vmYaml.vmName)
+				err = startFn()
+				Expect(err).ToNot(HaveOccurred())
+
+				By(fmt.Sprintf("Waiting for VM with %s api to become ready", vmYaml.apiVersion))
+
+				Eventually(func() bool {
+					virtualMachine, err := virtClient.VirtualMachine(tests.NamespaceTestDefault).Get(vmYaml.vmName, &metav1.GetOptions{})
+					Expect(err).ToNot(HaveOccurred())
+					if virtualMachine.Status.Ready {
+						return true
+					}
+					return false
+				}, 90*time.Second, 1*time.Second).Should(BeTrue())
+			}
+
+			// Update KubeVIrt from the previous release to the testing target release.
+			By("Updating KubeVirtObject With Current Tag")
+			patchKvVersionAndRegistry(kv.Name, curTag, curRegistry)
+
+			By("Wait for Updating Condition")
+			waitForUpdateCondition(kv)
+
+			By("Waiting for KV to stabilize")
+			waitForKv(kv)
+
+			By("Verifying infrastructure Is Updated")
+			allPodsAreReady(curTag)
+
+			// Verify console connectivity to VMI still works and stop VM
+			for _, vmYaml := range vmYamls {
+				By(fmt.Sprintf("Ensuring vm %s is ready and latest API annotation is set", vmYaml.apiVersion))
+				Eventually(func() bool {
+					// We are using our internal client here on purpose to ensure we can interact
+					// with previously created objects that may have been created using a different
+					// api version from the latest one our client uses.
+					virtualMachine, err := virtClient.VirtualMachine(tests.NamespaceTestDefault).Get(vmYaml.vmName, &metav1.GetOptions{})
+					Expect(err).ToNot(HaveOccurred())
+					if !virtualMachine.Status.Ready {
+						return false
+					}
+
+					return true
+				}, 90*time.Second, 1*time.Second).Should(BeTrue())
+
+				By(fmt.Sprintf("Connecting to %s's console", vmYaml.vmName))
+				vmi, err := virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Get(vmYaml.vmName, &metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				expecter, err := tests.LoggedInCirrosExpecter(vmi)
+				Expect(err).ToNot(HaveOccurred())
+				expecter.Close()
+
+				By("Stopping VM with virtctl")
+				stopFn := tests.NewRepeatableVirtctlCommand("stop", "--namespace", tests.NamespaceTestDefault, vmYaml.vmName)
+				Eventually(func() error {
+					return stopFn()
+				}, 30*time.Second, 1*time.Second).Should(BeNil())
+
+				By("Waiting for VMI to stop")
+				Eventually(func() bool {
+					_, err := virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Get(vmYaml.vmName, &metav1.GetOptions{})
+					if err != nil && errors.IsNotFound(err) {
```

Comment:
If the "err" is something other than "isNotFound", this is going to keep re-trying, but I imaging anything besides "IsNotFound" should probably trigger a test suite exception.

```
if err != nil {
    if !errors.IsNotFound(err) {
        Expect(err).ToNot(HaveOccurred())
    }
    return true
}
```

Reply:
some errors are eventually consistent. We care more if the errors are persistent across a time period.

Reply:
fixed. I think your way is more consistent with the assumptions we make elsewhere in the test.